index.js

```diff
@@ -64,7 +66,7 @@ AuthResource.prototype.initPassport = function() {
     var socialAuthCallback = function(token, tokenSecret, profile, done) {
         debug('Login callback - profile: %j', profile);
 
-        userCollection.store.first({socialAccountId: profile.id}, function(err, user) {
+        userCollection.store.first({socialAccountId: String(profile.id)}, function(err, user) {
```

Comment:
If you force-cast `profile.id` to `String` here, you'd also need to cast it when the user is created (line 80), because otherwise in the database there might be a `Number` and this might not find a match.

Please either remove this forced cast or add it to the place where `socialAccountId` is inserted into the database.